### PR TITLE
reduce to one rule tree

### PR DESCRIPTION
**Pull request by ppcad**

### CODE REVIEW

File: tests/unit/processor/dropper/test_dropper.py [hunk 1]
```diff
@@ -9,22 +9,17 @@
...
 
     @property
-    def specific_rules_dirs(self):
-        return self.CONFIG["specific_rules"]
-
-    @property
-    def generic_rules_dirs(self):
-        return self.CONFIG["generic_rules"]
+    def rules_dirs(self):
+        return self.CONFIG["rules"]
```
Comment (dtrai2) [inline tests/unit/processor/dropper/test_dropper.py:18]: I think you can delete this, it's already defined in the parent class.
```suggestion
```

File: tests/unit/processor/domain_label_extractor/test_domain_label_extractor.py [hunk 2]
```diff
@@ -10,18 +10,13 @@
...
 
     @property
-    def generic_rules_dirs(self):
-        return self.CONFIG.get("generic_rules")
-
-    @property
-    def specific_rules_dirs(self):
-        return self.CONFIG.get("specific_rules")
+    def rules_dirs(self):
+        return self.CONFIG.get("rules")
```
Comment (dtrai2) [inline tests/unit/processor/domain_label_extractor/test_domain_label_extractor.py:19]: I think you can delete this, it's already defined in the parent class.
```suggestion
```

File: tests/unit/processor/datetime_extractor/test_datetime_extractor.py [hunk 3]
```diff
@@ -13,17 +13,12 @@
...
 
     @property
-    def specific_rules_dirs(self):
-        return self.CONFIG.get("specific_rules")
-
-    @property
-    def generic_rules_dirs(self):
-        return self.CONFIG.get("generic_rules")
+    def rules_dirs(self):
+        return self.CONFIG.get("rules")
```
Comment (dtrai2) [inline tests/unit/processor/datetime_extractor/test_datetime_extractor.py:21]: I think you can delete this, it's already defined in the parent class.
```suggestion
```

File: tests/unit/processor/concatenator/test_concatenator.py [hunk 4]
```diff
@@ -10,18 +10,13 @@
...
 
     @property
-    def generic_rules_dirs(self):
-        return self.CONFIG["generic_rules"]
-
-    @property
-    def specific_rules_dirs(self):
-        return self.CONFIG["specific_rules"]
+    def rules_dirs(self):
+        return self.CONFIG["rules"]
```
Comment (dtrai2) [inline tests/unit/processor/concatenator/test_concatenator.py:19]: I think you can delete this, it's already defined in the parent class.
```suggestion
```

File: tests/unit/processor/generic_adder/test_generic_adder.py [hunk 5]
```diff
@@ -310,31 +310,26 @@ class TestGenericAdder(BaseProcessorTestCase):
...
 
     @property
-    def generic_rules_dirs(self):
-        return self.CONFIG.get("generic_rules")
-
-    @property
-    def specific_rules_dirs(self):
-        return self.CONFIG.get("specific_rules")
+    def rules_dirs(self):
+        return self.CONFIG.get("rules")
```
Comment (dtrai2) [inline tests/unit/processor/generic_adder/test_generic_adder.py:318]: I think you can delete this, it's already defined in the parent class.
```suggestion
```

File: tests/unit/processor/generic_resolver/test_generic_resolver.py [hunk 6]
```diff
@@ -27,18 +26,13 @@ class TestGenericResolver(BaseProcessorTestCase):
...
-        """Return the paths of the specific rules"""
-        return self.CONFIG["specific_rules"]
-
-    @property
-    def generic_rules_dirs(self):
-        """Return the paths of the generic rules"""
-        return self.CONFIG["generic_rules"]
+    def rules_dirs(self):
+        """Return the paths of the rules"""
+        return self.CONFIG["rules"]
```
Comment (dtrai2) [inline tests/unit/processor/generic_resolver/test_generic_resolver.py:31]: I think you can delete this, it's already defined in the parent class.
```suggestion
```

File: tests/unit/processor/geoip_enricher/test_geoip_enricher.py [hunk 7]
```diff
@@ -88,19 +88,14 @@ class TestGeoipEnricher(BaseProcessorTestCase):
...
 
     @property
-    def generic_rules_dirs(self):
-        return self.CONFIG["generic_rules"]
-
-    @property
-    def specific_rules_dirs(self):
-        return self.CONFIG["specific_rules"]
+    def rules_dirs(self):
+        return self.CONFIG["rules"]
```
Comment (dtrai2) [inline tests/unit/processor/geoip_enricher/test_geoip_enricher.py:98]: I think you can delete this, it's already defined in the parent class.
```suggestion
```

File: tests/unit/processor/labeler/test_labeler.py [hunk 8]
```diff
@@ -61,32 +61,26 @@ class TestLabeler(BaseProcessorTestCase):
...
+        "rules": ["tests/testdata/unit/labeler/rules"],
     }
 
     @property
-    def specific_rules_dirs(self):
-        """Return path to specific rule directories"""
-        return self.CONFIG["specific_rules"]
+    def rules_dirs(self):
+        """Return path to rule directories"""
+        return self.CONFIG["rules"]
```
Comment (dtrai2) [inline tests/unit/processor/labeler/test_labeler.py:70]: I think you can delete this, it's already defined in the parent class.
```suggestion
```

File: tests/unit/processor/template_replacer/test_template_replacer.py [hunk 9]
```diff
@@ -25,12 +24,8 @@ class TestTemplateReplacer(BaseProcessorTestCase):
...
 
     @property
-    def generic_rules_dirs(self):
-        return self.CONFIG.get("generic_rules")
-
-    @property
-    def specific_rules_dirs(self):
-        return self.CONFIG.get("specific_rules")
+    def rules_dirs(self):
+        return self.CONFIG.get("rules")
```
Comment (dtrai2) [inline tests/unit/processor/template_replacer/test_template_replacer.py:28]: I think you can delete this, it's already defined in the parent class.
```suggestion
```

**Review by dtrai2 — CHANGES_REQUESTED** [review]
Ah thanks for the diligent work! Quite a feat. It'll take out unnecessary the complexity.

I only found some tiny things you can still remove. Also in the following two files I found references to `generic` and `specific` rules:

- `rule.py` line 20 & 21
- `test_full_configuration.py` line 237 and 242

**Comment by ppcad** [comment]
> Ah thanks for the diligent work! Quite a feat. It'll take out unnecessary the complexity.
> 
> I only found some tiny things you can still remove. Also in the following two files I found references to `generic` and `specific` rules:
> 
> * `rule.py` line 20 & 21
> * `test_full_configuration.py` line 237 and 242

Thank you for the review, I've changed it.

**Review by dtrai2 — APPROVED** [review]

**Pull request merged**